lib/validator.js

```diff
@@ -149,130 +142,126 @@ module.exports = function validator(options) {
  * @returns {*}
  */
 function refresolver(schemas, value) {
-    var id, refschema, path, fragment, paths;
+  let id; let refschema; let path; let fragment; let
+    paths;
 
-    id = value.substr(0, value.indexOf('#') + 1);
-    path = value.substr(value.indexOf('#') + 1);
+  id = value.substr(0, value.indexOf('#') + 1);
+  path = value.substr(value.indexOf('#') + 1);
 
-    if (id) {
-        refschema = schemas[id] || schemas[id.substr(0, id.length - 1)];
-    }
-    else {
-        refschema = schemas['#'];
-    }
+  if (id) {
+    refschema = schemas[id] || schemas[id.substr(0, id.length - 1)];
+  } else {
+    refschema = schemas['#'];
+  }
 
-    assert.ok(refschema, 'Can not find schema reference: ' + value + '.');
+  assert.ok(refschema, `Can not find schema reference: ${value}.`);
 
-    fragment = refschema;
-    paths = Array.isArray(path) ? path : path.split('/');
+  fragment = refschema;
+  paths = Array.isArray(path) ? path : path.split('/');
 
-    for (var i = 1; i < paths.length && fragment; i++) {
-        fragment = typeof fragment === 'object' && fragment[paths[i]];
-    }
+  // eslint-disable-next-line no-plusplus
+  for (let i = 1; i < paths.length && fragment; i++) {
```

Comment:
```suggestion
  for (let i = 1; i < paths.length && fragment; i+=1) {
```
We could use the `i += 1` expression or we could add the `"allowForLoopAfterthoughts": true` rule to the eslint config file.

Reply:
sure, the change is done.